packages/report-server/src/routes/TestReportRoute.ts

```diff
@@ -22,11 +22,43 @@ export type TestReportRequest = Request<
   ReportRouteQuery
 >;
 
+const BES_DATA_ADMIN_PERMISSION_GROUP = 'BES Data Admin';
+
 export class TestReportRoute extends Route<TestReportRequest> {
+  async checkUserHasAccessToReport(hierarchy = 'explore', requestedOrgUnitCodes: string[]) {
+    const { accessPolicy, ctx } = this.req;
+
+    const foundOrgUnits = await ctx.microServices.entityApi.getEntities(
+      hierarchy,
+      requestedOrgUnitCodes,
+      {
+        fields: ['code', 'country_code'],
+      },
+    );
+    const foundOrgUnitCodes = foundOrgUnits.map(orgUnit => orgUnit.code);
+
+    const missingOrgUnitCodes = requestedOrgUnitCodes.filter(
+      orgUnitCode => !foundOrgUnitCodes.includes(orgUnitCode),
+    );
+    if (missingOrgUnitCodes.length > 0) {
+      throw new Error(`No entities found with codes ${missingOrgUnitCodes}`);
+    }
+
+    const countryCodes = new Set(foundOrgUnits.map(orgUnit => orgUnit.country_code));
+    countryCodes.forEach(countryCode => {
+      if (
+        countryCode === null ||
+        !accessPolicy.allows(countryCode, BES_DATA_ADMIN_PERMISSION_GROUP)
+      ) {
+        throw new Error(`No ${BES_DATA_ADMIN_PERMISSION_GROUP} access for user to ${countryCode}`);
+      }
+    });
+  }
```

Comment:
@billli0 depending on in this PR of your PR: https://github.com/beyondessential/tupaia/pull/3061 goes in first, one of us should refactor this route and the `FetchReportRoute` to share this common code, as they have a lot of similarity now.

Reply:
Thanks for letting me know @rohan-bes 🙌